pages/lecteur/cnaf/beneficiaire/index.js

```diff
@@ -95,14 +96,45 @@ export default function Beneficiaire() {
       const items = new Array(...dom.getElementsByTagName('InfosFoyerRSA'))
 
       setIsPending(false);
+// Comme je ne sais pas si c'est clair pour toi je l'écris, il faut faire attention car TOPPERSDRODEVORSA est associé à des personnes et ETATDOSRSA aux dossiers. Il me semble important de séparer les décomptes pour éviter de sommer des choux et des carottes.
+
+// Comme dit par téléphone, j'aurais laisser ces stats dans le tableau initial ou bien dans un second tableau mais qui affiche les valeurs pour tous les fichiers testés.
+
+// Note pour plus tard :
+
+// Il pourra être intéressant, dans une seconde itération de recouper TOPPERSDRODEVORSA avec ROLEPERS.
 
+// Il pourra être intéressant dans une troisième itération de faire le croisement entre ETATDOSRSA d'une part et ( TOPPERSDRODEVORSA / ROLEPERS ) d'autres part mais pour cela il faudra un tableau avec une ligne par valeur de ETATDOSRSA et une colonne par croisement ( TOPPERSDRODEVORSA / ROLEPERS ).
       const dates = items
         .map(i => i.getElementsByTagName('DTDEMRSA')[0].innerHTML)
         .reduce((accum, value) => {
           accum[value] = (accum[value] || 0 ) + 1
           return accum
         }, {})
 
+      const processField = (accum, value) => {
+          if (value[0]) {
+            accum[value[0].innerHTML] = (accum[value[0].innerHTML] || 0) + 1
+            accum["Total"] = (accum["Total"] || 0) + 1
+          }
+          return accum
+        }
+
+      const devoirs = items
+        .map(i => i.getElementsByTagName('TOPPERSDRODEVORSA'))
+        .reduce(processField, {})
+        console.log("Start")
+      let newKeysDevoirs = keysDevoirs.concat(Object.keys(devoirs));
+      newKeysDevoirs = Array.from(new Set(newKeysDevoirs))
+      setKeysDevoirs(newKeysDevoirs)
```

Comment:
Je ne comprends pas pourquoi tout ça est nécessaire.

Reply:
Pour le calcul du nombre de colonnes dans la view

Reply:
`Object.keys(devoirs)` et `Object.keys(devoirs).length` ne donnent pas la réponse directement ?

Reply:
Object.keys(devoirs) marche quand on est à l'intérieur d'un run, or le tableau de stats affiche les stats de tous les fichiers qui ont été analysés, d'où le concat et l'utilisation d'un state keysDevoirs et keysDroits.

Reply:
Ok, merci, ça pique un peu les yeux quand même.

Est ce que c'est mieux ça, je ne sais pas trop.
```suggestion
      const newKeysDevoirs = keysDevoirs.concat(Object.keys(devoirs));
      setKeysDevoirs(Array.from(new Set(newKeysDevoirs)))
```